### PR TITLE
[Authorize upgrade] Upgrade authorization by Polkadot relay chain

### PR DESCRIPTION
**Pull request by karolk91**
This pull request introduces scenarios of Runtime upgrades authorization by Relay Chain.

Scenarios to cover:
- [x] Polkadot relay chain authorizes upgrade for itself
- [x] Polkadot relay chain authorizes upgrade for Polkadot Asset Hub
- [x] Polkadot relay chain authorizes upgrade for Polkadot Bridge Hub
- [x] Polkadot relay chain authorizes upgrade for Polkadot Coretime
- [x] Polkadot relay chain authorizes upgrade for Polkadot People
- [x] Polkadot relay chain authorizes upgrade for Polkadot Collectives

This pull request covers Polkadot Network, other PRs will be created for Kusama network and AssetHub Next based scenarios

### CODE REVIEW

**Comment by karolk91** [comment]
hey @xlc , @rockbmb 

I would like to discuss with you the approach I used here, that I'm not sure if it looks nice.

You will notice that I've introduced test file that involve scenarios for 3 chains, and following current pattern of naming files I'm using:
`chain1.chain2.chain3.test.ts`

the truth is that chain3 is always Collectives, so more like
`chain1.chain2.collectives.test.ts`

I was wondering if that looks ok to you, or should I ignore Collectives part (as something that is fixed among tests, but will probably be skipped for notifications purposes) ?

Also, I have in mind a test that will involve all the system chains and relay chain, so something like an edge case scenario where relay chain approves upgrade for all system chains at once - how to fit that scenario? should that be part of `polkadot.test.ts` or maybe `polkadot.allSystem.test.ts`, how do you see it?

File: packages/shared/src/upgrade.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    proofSize: '10000',
+    refTime: '500000000',
+  })
+  await collectivesChain.dev.newBlock()
+  await checkSystemEvents(collectivesChain, 'polkadotXcm')
+    .redact({ hash: false, redactKeys: /messageId/ })
+    .toMatchSnapshot('collectives events emitted when sending xcm')
+  await governingChain.dev.newBlock()
+  await checkSystemEvents(governingChain, 'whitelist', 'messageQueue')
+    // TODO: redacting 'id' is not perfect here; but seems not possible to redact nested keys, in this case specifically only "message.id"
```
Comment (xlc) [inline packages/shared/src/upgrade.ts:124]: you can do `.map(x => { x.message.id = '(redacted)'; return x }`
or I can add a `redactKeyPaths: string[][]` and basically `_.set(x, keyPath, '(redacted)')` to the object

**Comment by xlc** [comment]
`chain1.chain2.chain3.test.ts` is fine. we can easily update the notification to support such case.
for the one with all the system chains, I will say just make it `polkadot.test.ts`

File: packages/shared/src/upgrade.ts (hunk 1 above)
Comment (karolk91) [reply packages/shared/src/upgrade.ts]: thank you for looking into this, seems I must have confused/overlooked something in this case as I couldn't justify this again on second look - `id` is properly redacted here, no conflicts with other nested `id` attributes

I have removed this misleading comment - nothing to be done here

**Review by xlc — APPROVED** [review]
